src/parser/command.rs

```diff
@@ -213,6 +220,42 @@ pub fn parse_defframe<'a>(input: ParserInput<'a>) -> ParserResult<'a, Instructio
     ))
 }
 
+/// Parse the contents of a `DEFGATE` instruction.
+pub fn parse_defgate<'a>(input: ParserInput<'a>) -> ParserResult<'a, Instruction> {
+    let (input, name) = token!(Identifier(v))(input)?;
+    let (input, parameters) = opt(delimited(
+        token!(LParenthesis),
+        separated_list1(token!(Comma), token!(Variable(v))),
+        token!(RParenthesis),
+    ))(input)?;
+    let (input, gate_type) = opt(preceded(
+        token!(As),
+        alt((
```

Comment:
Should use the `wrapped_parsers` version of `alt`, which gives a much better error message.

Reply:
That version expects a `LexInput`, not a `ParserInput`. It's specific to lexing.